json_patch.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2021 Alexadru Ardelean.
+ *
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the MIT license. See COPYING for details.
+ *
+ */
+
+/**
+ * @file
+ * @brief JSON Patch (RFC 6902) implementation for manipulating JSON objects
+ */
+#ifndef _json_patch_h_
+#define _json_patch_h_
+
+#include "json_pointer.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Apply the JSON patch to the base object.
+ * The patch object must be formatted as per RFC 6902.
+ * If the patch is not correctly formatted, an error will
+ * be returned.
+ *
+ * The original `base` object will first be copied, and then
```

Comment:
As I mentioned on the json_object_deep_copy() call in json_patch.c, let's leave this up to the caller to do or not.